File: backend/app/schemas/fees.py
from pydantic import BaseModel
from typing import Dict, List


class FeeBreakdown(BaseModel):
    fees: Dict[str, float]
    subtotal: float
    final_amount: float


class StudentFeeDetail(BaseModel):
    student_id: str
    student_name: str
    fee_breakdown: FeeBreakdown


class DetailedFeeCalculationResponse(BaseModel):
    total_amount: float
    student_fees: List[StudentFeeDetail]

File: mobile-app/src/api/fees.ts
import { post } from './client';

export type FeeCalculationResponse = {
  total_amount: number;
  student_fees: {
    student_id: string;
    student_name: string;
    fee_breakdown: {
      fees: Record<string, number>;
      subtotal: number;
      final_amount: number;
      discount_amount?: number;
    };
  }[];
};

export const FeesApi = {
  calculate(student_ids: string[]) {
    return post<FeeCalculationResponse>(`/api/fees/calculate-fees`, {
      student_ids,
      student_club_ids: {},
    });
  },
};